src/python/pants/backend/project_info/filter_targets.py

```diff
@@ -136,3 +139,22 @@ def warn_deprecated_target_type(tgt_type: type[Target]) -> None:
         entity=f"using `--filter-target-type={tgt_type.deprecated_alias}`",
         hint=f"Use `--filter-target-type={tgt_type.alias}` instead.",
     )
+
+
+class FilterGoal(Goal):
```

Comment:
A TODO here should refer back to the special case in the options system for having this installed.